avocado/core/job.py

```diff
@@ -422,8 +422,7 @@ def _run(self):
         self._setup_job_results()
         self.__start_job_logging()
 
-        if (getattr(self.args, 'remote_hostname', False) and
-           getattr(self.args, 'remote_no_copy', False)):
+        if getattr(self.args, 'remote_execution', False):
```

Comment:
The beauty is that we don't need any checks there. The `remote` plugins can directly override the `args.loaders`.

Similarly we can abuse the `Mux.parsed` to disable `yaml/mux` processing in case the `args.mux` is already parsed.

With those two we would not need any checks for `remote_execution` at all :wink:. Anyway as I said before this solution is better then previous situation so if you don't want to go this way you can keep it as is (here)